build/Generator/ApiGenerator.php

```diff
@@ -119,6 +127,116 @@ public function generateResultClass(array $shapes, $service, $baseNamespace, $cl
+        $constructorBody = '';
+
+        foreach ($members as $name => $data) {
+            $parameterType = $members[$name]['shape'];
+            $memberShape = $shapes[$parameterType];
+            $nullable = true;
+            if ('structure' === $memberShape['type']) {
+                $this->generateInputClass($definition, $service, $operationName, $baseNamespace, $parameterType);
+                $returnType = $baseNamespace . '\\' . $parameterType;
+                $constructorBody .= sprintf('$this->%s = isset($input["%s"]) ? %s::create($input["%s"]) : null;' . "\n", $name, $name, $parameterType, $name);
+            } elseif ('list' === $memberShape['type']) {
+                $parameterType = $memberShape['member']['shape'] . '[]';
+                $this->generateInputClass($definition, $service, $operationName, $baseNamespace, $memberShape['member']['shape']);
+                $returnType = $baseNamespace . '\\' . $memberShape['member']['shape'];
+                $constructorBody .= sprintf('$this->%s = array_map(function($item) { return %s::create($item); }, $input["%s"] ?? []);' . "\n", $name, $memberShape['member']['shape'], $name);
+                $nullable = false;
+            } else {
+                $returnType = $parameterType = $this->toPhpType($memberShape['type']);
+                $constructorBody .= sprintf('$this->%s = $input["%s"] ?? null;' . "\n", $name, $name);
+            }
+
+            $property = $class->addProperty($name)->setPrivate();
+            if (\in_array($name, $shapes[$className]['required'] ?? [])) {
+                $property->addComment('@required');
+            }
+            $property->addComment('@var ' . $parameterType . ($nullable ? '|null' : ''));
+
+            $returnType = '[]' === substr($parameterType, -2) ? 'array' : $returnType;
+
+            $class->addMethod('get' . $name)
+                ->setReturnType($returnType)
+                ->setReturnNullable($nullable)
+                ->setBody(
+                    <<<PHP
+return \$this->{$name};
+PHP
+                );
+
+            $class->addMethod('set' . $name)
+                ->setReturnType('self')
+                ->setBody(
+                    <<<PHP
+\$this->{$name} = \$value;
+return \$this;
+PHP
+                )
+                ->addParameter('value')->setType($returnType)->setNullable($nullable)
+            ;
+        }
+
+        $constructor->setBody($constructorBody);
+        if ($root) {
+            $this->inputClassRequestGetters($inputShape, $class);
+        }
+
+        $printer = new PsrPrinter();
+        \file_put_contents(\sprintf('%s/%s/Input/%s.php', $this->srcDirectory, $service, $className), "<?php\n\n" . $printer->printNamespace($namespace));
+    }
+
+
+    private function inputClassRequestGetters(array $inputShape, ClassType $class): void
+    {
+        foreach (['header' => '$headers', 'querystring' => '$query', 'uri' => '$uri'] as $locationName => $varName) {
```

Comment:
URI have a special meaning (https://github.com/aws/aws-sdk-php/blob/master/src/data/s3/2006-03-01/api-2.json#L73).
There are placeholders for the parameter http.requestUri.
See my PR

Reply:
Thank you. 
I did not pay much attention to the URI

Reply:
I think I've rebased this properly